core/src/test/java/io/github/tcdl/msb/api/ChannelMonitorIT.java

```diff
@@ -37,10 +58,158 @@ public void testAnnouncement() throws InterruptedException {
+                payload);
+        MockAdapter.pushRequestMessage(requestMessage.getTopics().getResponse(), Utils.toJson(responseMessage, msbContext.getPayloadMapper()));
+
+        assertTrue("Heartbeat response was not received",
+                heartBeatResponseReceived.await(RequesterResponderIT.MESSAGE_TRANSMISSION_TIME, TimeUnit.MILLISECONDS));
+    }
+
+    @Test
+    public void testHeartbeatUnexpectedMessage() throws InterruptedException {
+        String TOPIC_NAME = "topic2";
+
+        Map<String, AgentTopicStats> topicInfoMap = new HashMap<>();
+        Instant lastProduced = Instant.now().minusMillis(20000);
+        Instant lastConsumed = Instant.now().minusMillis(30000);
+        topicInfoMap.put(TOPIC_NAME, new AgentTopicStats(true, false, lastProduced, lastConsumed));
+
+        Payload payload = new Payload.Builder<Object, Object, Object, Map<String, AgentTopicStats>>()
+                .withBody(topicInfoMap)
+                .build();
+
+        CountDownLatch heartBeatResponseReceived = new CountDownLatch(1);
+        Callback<AggregatorStats> handler = stats -> {
+            assertEquals(1, stats.getTopicInfoMap().size());
+            heartBeatResponseReceived.countDown();
+        };
+
+        channelMonitorAggregator = msbContext.getObjectFactory().createChannelMonitorAggregator(handler);
+        channelMonitorAggregator.start(true, ChannelMonitorAggregator.DEFAULT_HEARTBEAT_INTERVAL_MS, 1000);
+
+        //need to await for original request for heartbeat to be send to simulate response with same correlationId
+        Message requestMessage = awaitHeartBeatRequestSent(RequesterResponderIT.MESSAGE_TRANSMISSION_TIME);
+
+        Message brokenResponseMessage = createMsbRequestMessage(requestMessage.getTopics().getResponse(), requestMessage.getCorrelationId(),
+                " unexpected statistics format received");
+        Message responseMessage = createMsbRequestMessage(requestMessage.getTopics().getResponse(), requestMessage.getCorrelationId(),
+                payload);
+
+        MockAdapter.pushRequestMessage(requestMessage.getTopics().getResponse(), Utils.toJson(brokenResponseMessage, msbContext.getPayloadMapper()));
+        MockAdapter.pushRequestMessage(requestMessage.getTopics().getResponse(), Utils.toJson(responseMessage, msbContext.getPayloadMapper()));
+        MockAdapter.pushRequestMessage(requestMessage.getTopics().getResponse(), Utils.toJson(brokenResponseMessage, msbContext.getPayloadMapper()));
+
+        assertTrue("Heartbeat response was not received",
+                heartBeatResponseReceived.await(RequesterResponderIT.MESSAGE_TRANSMISSION_TIME, TimeUnit.MILLISECONDS));
+    }
+
+    private Message awaitHeartBeatRequestSent(long timeoutMs) {
+        LocalDateTime startTime = LocalDateTime.now();
+        String originalRequest = null;
+        while (originalRequest == null && (LocalDateTime.now().isBefore(startTime.plus(timeoutMs, ChronoUnit.MILLIS)))) {
```

Comment:
1st - don't understand.
2 extension to MockAdapter. We dont use any cosnumer-requester there..just operate on Maps directly.this is just a version of MockAdapter.pollJsonMessageForTopic with await.